checkmultimemberships.py

```diff
@@ -0,0 +1,138 @@
+    # ----------------------------------------------------------------------
+    multimembers = {}
+    multimemberdata = {}
+    multimembersinglecnt = 0
+
+    with open(membercachefile, 'r') as memfile:
+        today = datetime.now()
+        cachedmembers = DictReader(memfile)
+        for memberrec in cachedmembers:
+            enddate = ymd.asc2dt(memberrec['ExpirationDate'])
+
+            if (((memberrec['MembershipType'] == "Family") or
+                (memberrec['MembershipType'] == "Two in Household") or
+                (memberrec['MembershipType'] == "Individual + Junior")) and
+                (enddate >= today)):
+                if ((memberrec['MembershipID']) in multimembers):
+                    multimembers[memberrec['MembershipID']] += 1
+                    if ((memberrec['MembershipID']) in multimemberdata):
+                        del multimemberdata[memberrec['MembershipID']]
+                else:
+                    multimembers[memberrec['MembershipID']] = 1
+                    multimemberdata[memberrec['MembershipID']] = memberrec
+
+    if multifile:
+        with open(multifile, 'w', newline='') as multif:
+            cachehdr = 'MemberID,MembershipID,MembershipType,FamilyName,GivenName,MiddleName,Gender,DOB,Email,PrimaryMember,JoinDate,ExpirationDate,LastModified'.split(
+                ',')
+            cache = DictWriter(multif, cachehdr)
+            cache.writeheader()
+            for memberid, memberCnt in multimembers.items():
+                if memberCnt == 1:
+                    multimembersinglecnt += 1
+                    # find the member record that corresponds to memberid
+                    memberrec = multimemberdata[memberid]
+                    if (memberrec):
+                        cache.writerow(memberrec)
+
+    return multimembersinglecnt
+
+# ----------------------------------------------------------------------
+def summarize(configfile, debug=False):
+    # ----------------------------------------------------------------------
+    '''
+    Summarize the membership stats and members for a given runsignup club.
+
+    :param configfile: configuration filename
+    :param debug: True for requests debugging
+    '''
+
+    # configuration file supplied -- pull credentials from the app section
+    appconfig = getitems(configfile, 'runsignup')
+    club = appconfig['RSU_CLUB']
+    membercachefile = appconfig['RSU_CACHEFILE']
+    memberstatsfile = appconfig['RSU_STATSFILE']
+    membermultifile = appconfig['RSU_MULTIFILE']
+    key = appconfig['RSU_KEY']
+    secret = appconfig['RSU_SECRET']
+
+    # update member cache file
+    # note this update is done under lock to prevent any apache thread from disrupting it
+#    members = updatemembercache(club, membercachefile, key=key, secret=secret, debug=debug)
+
+    # analyze the memberships for multi-person memberships with only one member
+    multicount = analyzemultimembers(membercachefile, multifile=membermultifile)

```

Comment:
This assumes membercachefile is available. maybe that's ok. I need to think about how this reporting function would fit into a larger system. Currently the membercachefile is under steeplechasers.org, but if we build a membertility app, it would need to be maintained in two places. Which may not be terrible. Again, we need to think about how this would fit into a larger system.

Initially I was thinking the input data would be grabbed directly from RunSignUp, not via the cache file.